h5py/tests/test_dtype.py

```diff
@@ -492,6 +493,23 @@ def _test_b8(self, arr1, expected_default_cast_dtype, cast_dtype=None):
                 arr3 = dset[:]
             self.assertArrayEqual(arr3, arr1.astype(cast_dtype, copy=False))
 
+    def test_b16_uint16(self):
+        arr1 = np.array(np.arange(10), dtype=np.dtype(np.uint16))
+        path = self.mktemp()
+        with h5py.File(path, 'w') as f:
+            space = h5py.h5s.create_simple(arr1.shape)
+            dset_id = h5py.h5d.create(f.id, b'test', h5py.h5t.STD_B16LE, space)
+            dset = h5py.Dataset(dset_id)
+            dset[:] = arr1
+
+        with h5py.File(path, 'r') as f:
+            dset = f['test']
+            arr2 = dset[:]
+            self.assertArrayEqual(
+                arr2,
+                arr1.astype(np.dtype(np.uint16), copy=False)
+            )
```

Comment:
```suggestion
            self.assertArrayEqual(dset[:], arr1)
```

arr1 already has dtype uint16, so no need to convert it - unless I'm missing something?